src/Microsoft.Health.Fhir.Core/Messages/Operation/EverythingOperationRequest.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Messages.Operation
+{
+    public class EverythingOperationRequest : IRequest<EverythingOperationResponse>
+    {
+        public EverythingOperationRequest(string resourceType, string resourceId, PartialDateTime start = null, PartialDateTime end = null, PartialDateTime since = null, string type = null, int? count = null, string continuationToken = null)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(resourceType, nameof(resourceType));
+            EnsureArg.IsNotNullOrWhiteSpace(resourceId, nameof(resourceId));
+
+            ResourceType = resourceType;
+            ResourceId = resourceId;
+            Start = start;
+            End = end;
+            Since = since;
+            Type = type;
+            Count = count;
+            ContinuationToken = continuationToken;
+        }
+
+        public string ResourceType { get; }
+
+        public string ResourceId { get; }
+
+        public PartialDateTime Start { get; }
+
+        public PartialDateTime End { get; }
+
+        public PartialDateTime Since { get; }
+
+        public string Type { get; }
+
+        public int? Count { get; }
+
+        public string ContinuationToken { get; }
+
+        public IReadOnlyList<string> Includes { get; } = new[] { "general-practitioner", "organization" };
+
+        public IReadOnlyList<Tuple<string, string>> Revincludes { get; } = new[] { Tuple.Create("Device", "patient") };
```

Comment:
Why these ones, how they been determined?

Reply:
Also consider what this is readonly lists, wouldn't it be simplier to define them in handler and pass to search, than define them here?

This is just message, it feels strange to hide implementation details in it.